mlflow/sparkml.py

```diff
@@ -0,0 +1,122 @@
+    :param conda_env: Path to a Conda environment file. If provided, this defines enrionment for
+           the model. At minimum, it should specify python, pyspark and mlflow with appropriate
+           versions.
+    :param jars: List of jars needed by the model.
+    """
+    return Model.log(artifact_path=artifact_path, flavor=mlflow.sparkml, spark_model=spark_model,
+                     jars=jars, env=conda_env)
+
+
+def save_model(spark_model, path, mlflow_model=Model(), conda_env=None, jars=None):
+    """
+    Save Spark MLlib PipelineModel at given local path.
+
+    Uses Spark MLlib persistence mechanism.
+
+    :param spark_model: Spark PipelineModel to be saved. Currently can only save PipelineModels.
+    :param path: Local path where the model is to be saved.
+    :param mlflow_model: MLflow model config this flavor is being added to.
+    :param conda_env: Conda environment this model depends on.
+    :param jars: List of jars needed by the model.
+    """
+    if jars:
+        raise Exception("jar dependencies are not implemented")
+    if not isinstance(spark_model, Transformer):
+        raise Exception("Unexpected type {}. SparkML model works only with Transformers".format(
+            str(type(spark_model))))
+    if not isinstance(spark_model, PipelineModel):
+        raise Exception("Not a PipelineModel. SparkML can currently only save PipelineModels.")
+    spark_model.save(os.path.join(path, "model"))
+    pyspark_version = pyspark.version.__version__
+    model_conda_env = None
+    if conda_env:
+        model_conda_env = os.path.basename(os.path.abspath(conda_env))
+        shutil.copyfile(conda_env, os.path.join(path, model_conda_env))
+    if jars:
+        raise Exception("jar dependencies are not yet implemented")
+    mlflow_model.add_flavor('sparkml', pyspark_version=pyspark_version, model_data="model")
+    pyfunc.add_to_model(mlflow_model, loader_module="mlflow.sparkml", data="model",
+                        env=model_conda_env)
+    mlflow_model.save(os.path.join(path, "MLmodel"))
+
+
+def load_model(path, run_id=None):
+    """
+    Load the Spark MLlib model from the given path.
+    :param run_id: Run ID. If provided it is combined with path to identify the model.
+    :param path: Local filesystem path or Run-relative artifact path to the model.
+    :return: SparkML model.
+    :rtype: pyspark.ml.pipeline.PipelineModel
+    """
+    if run_id is not None:
+        path = mlflow.tracking._get_model_log_dir(model_name=path, run_id=run_id)
+    m = Model.load(path)
+    if FLAVOR_NAME not in m.flavors:
+        raise Exception("Model does not have {} flavor".format(FLAVOR_NAME))
+    conf = m.flavors[FLAVOR_NAME]
+    return PipelineModel.load(conf['model_data'])
+
+
+def load_pyfunc(path):
+    """
+    Load the model as PyFunc.
+    :param path: Local path
+    :return: The model as PyFunc.
+    """
+    spark = pyspark.sql.SparkSession.builder.config(key="spark.python.worker.reuse", value=True) \
```

Comment:
Nit: do we need to specify keyword args `key` and `value` here? It looks a bit weird.